crates/uv/src/commands/project/version.rs

```diff
@@ -0,0 +1,597 @@
+                )
+            }
+        }
+        err => {
+            anyhow!("{err}: {}", pyproject_path.user_display())
+        }
+    })?;
+
+    // Figure out new metadata
+    let new_version = if let Some(value) = value {
+        match Version::from_str(&value) {
+            Ok(version) => Some(version),
+            Err(err) => match &*value {
+                "major" | "minor" | "patch" => {
+                    return Err(anyhow!(
+                        "Invalid version `{value}`, did you mean to pass `--bump {value}`?"
+                    ));
+                }
+                _ => {
+                    return Err(err)?;
+                }
+            },
+        }
+    } else if let Some(bump) = bump {
+        Some(bumped_version(&old_version, bump, printer)?)
+    } else {
+        None
+    };
+
+    // Update the toml and lock
+    let status = if dry_run {
+        ExitStatus::Success
+    } else if let Some(new_version) = &new_version {
+        let project = update_project(project, new_version, &mut toml, &pyproject_path)?;
+        Box::pin(lock_and_sync(
+            project,
+            project_dir,
+            locked,
+            frozen,
+            active,
+            no_sync,
+            python,
+            install_mirrors,
+            &settings,
+            network_settings,
+            python_preference,
+            python_downloads,
+            installer_metadata,
+            concurrency,
+            no_config,
+            cache,
+            printer,
+            preview,
+        ))
+        .await?
+    } else {
+        debug!("No changes to version; skipping update");
+        ExitStatus::Success
+    };
+
+    // Report the results
+    let old_version = VersionInfo::new(name.as_ref(), &old_version);
+    let new_version = new_version.map(|version| VersionInfo::new(name.as_ref(), &version));
+    print_version(old_version, new_version, short, output_format, printer)?;
+
+    Ok(status)
+}
+
+/// Find the pyproject.toml we're modifying
+///
+/// Note that `uv version` never needs to support PEP723 scripts, as those are unversioned.
+async fn find_target(project_dir: &Path, package: Option<&PackageName>) -> Result<VirtualProject> {
+    // Find the project in the workspace.
+    // No workspace caching since `uv version` changes the workspace definition.
+    let project = if let Some(package) = package {
+        VirtualProject::Project(
+            Workspace::discover(
+                project_dir,
+                &DiscoveryOptions::default(),
+                &WorkspaceCache::default(),
+            )
+            .await?
+            .with_current_project(package.clone())
+            .with_context(|| format!("Package `{package}` not found in workspace"))?,
+        )
+    } else {
+        VirtualProject::discover(
+            project_dir,
+            &DiscoveryOptions::default(),
+            &WorkspaceCache::default(),
+        )
+        .await?
+    };
```

Comment:
This is how we find the project we lookup the pyproject.toml to get the name from